crates/utils/sov-soak-lib/src/lib.rs

```diff
@@ -0,0 +1,466 @@
+use std::time::Duration;
+
+use rand::Rng;
+use sov_bank::Bank;
+use sov_bank::CallMessageDiscriminants::Transfer;
+use sov_mock_da::BlockProducingConfig;
+use sov_modules_api::capabilities::config_chain_id;
+use sov_modules_api::macros::config_value;
+use sov_modules_api::prelude::arbitrary::{self, Unstructured};
+use sov_modules_api::prelude::tracing;
+use sov_modules_api::transaction::TxDetails;
+use sov_modules_api::{Amount, CryptoSpec, DispatchCall, EncodeCall, PrivateKey, Runtime, Spec};
+use sov_paymaster::{
+    PayeePolicy, PayerGenesisConfig, Paymaster, PaymasterConfig, PaymasterPolicyInitializer,
+    SafeVec,
+};
+use sov_sequencer::preferred::PreferredSequencerConfig;
+use sov_sequencer::SequencerKindConfig;
+use sov_synthetic_load::CallMessageDiscriminants::{
+    ReadAndSetHeavyState, ReadAndSetManyIndividualValues, RunCPUHeavyOperation,
+};
+use sov_synthetic_load::SyntheticLoad;
+use sov_test_utils::runtime::genesis::zk::config::HighLevelZkGenesisConfig;
+use sov_test_utils::runtime::genesis::zk::MinimalZkGenesisConfig;
+use sov_test_utils::test_rollup::{GenesisSource, RollupBuilder, TestRollup};
+use sov_test_utils::{
+    generate_runtime, RtAgnosticBlueprint, TestProver, TestSequencer, TestSpec, TestUser,
+    TransactionType, TEST_DEFAULT_MAX_FEE, TEST_DEFAULT_MAX_PRIORITY_FEE,
+    TEST_DEFAULT_USER_BALANCE,
+};
+use sov_transaction_generator::generators::bank::harness_interface::BankHarness;
+use sov_transaction_generator::generators::bank::BankMessageGenerator;
+use sov_transaction_generator::generators::basic::{
+    BasicCallMessageFactory, BasicChangeLogEntry, BasicModuleRef, BasicTag,
+};
+use sov_transaction_generator::generators::synthetic_load::{
+    SyntheticLoadHarness, SyntheticLoadMessageGenerator,
+};
+use sov_transaction_generator::interface::rng_utils::{get_random_bytes, randomize_buffer};
+use sov_transaction_generator::interface::MessageValidity;
+use sov_transaction_generator::{Distribution, GeneratedMessage, Percent, State};
+use tokio::sync::watch::Receiver;
+
+pub const DEFAULT_BLOCK_TIME_MS: u64 = 200;
+pub const DEFAULT_BLOCK_PRODUCING_CONFIG: BlockProducingConfig = BlockProducingConfig::Periodic {
+    block_time_ms: DEFAULT_BLOCK_TIME_MS,
+};
+
+pub type TestRT = TestRuntime<TestSpec>;
+pub type RollupBlueprint = RtAgnosticBlueprint<TestSpec, TestRT>;
+pub type TestRollupBuilder = RollupBuilder<RollupBlueprint, PathBuf>;
+
+pub const DEFAULT_FINALIZATION_BLOCKS: u32 = 5;
+
+generate_runtime! {
+    name: TestRuntime,
+    modules: [paymaster: Paymaster<S>, synthetic_load: SyntheticLoad<S>],
+    operating_mode: sov_modules_api::runtime::OperatingMode::Zk,
+    minimal_genesis_config_type: MinimalZkGenesisConfig<S>,
+    gas_enforcer: paymaster: Paymaster<S>,
+    runtime_trait_impl_bounds: [],
+    kernel_type: sov_kernels::soft_confirmations::SoftConfirmationsKernel<'a, S>,
+    auth_type: sov_modules_api::capabilities::RollupAuthenticator<S, Self>,
+    auth_call_wrapper: |call| call,
+}
```

Comment:
What if we make sov-soak-lib completely runtime-free and specific soak tests supply their own runtimes, for separation of concerns? I.e. move the TestRuntime definition back to the demo rollup soak tests.
That way if you import the lib somewhere else to test your own rollup, it doesn't generate a TestRuntime that you don't need.

Though if it doesn't affect dependencies much then this might not mattter.

Reply:
Good point!